ouroboros-consensus/src/Ouroboros/Consensus/Util/Counting.hs

```diff
@@ -66,6 +69,19 @@ import           Ouroboros.Consensus.Util.SOP
 
 newtype Exactly xs a = Exactly { getExactly :: NP (K a) xs }
 
+-- NOTE: This could be a deriving newtype instance, but the constraints will be
+-- a bit more complicated.
+instance forall xs a . (Typeable xs, ToCBOR a, SListI xs) => ToCBOR (Exactly xs a) where
```

Comment:
Similar to my question on the `NP` instance; maybe use `hfoldMap`?

Reply:
Coincidence that you named it `Top` --- there is https://hackage.haskell.org/package/sop-core-0.5.0.1/docs/Data-SOP-Constraint.html#t:Top you can use with `hcfoldMap` if you'd like.

Reply:
Ahhh interesting. Another SOP nugget of knowledge there. I'll make use of that here.